### DISCUSSION TITLE
[Feature] Scheduled Backup Time for immich Photo Backup App

### DISCUSSION THREAD

**Discussion by withJihyuk** [immich-app/immich#9008] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

hello
It may be that I didn't check the function or didn't see the related suggestion, but I'd like to suggest it! I think it would be a good idea to create a function that allows backups only during the times reserved by the user. In my case, I accessed and made a reservation through VPN, but I had difficulty with automatic backup because it was not turned on all the time. Therefore, I make a suggestion.

thank you

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by devtechk** [comment]
This is a nice question, I notice more drain battery on iPhone, it could be improvement.
Thank you immich!

**Comment by homekepler-maker** [comment]
To add a minor item to this feature request since my new one was a duplicate 😅

Extended Intervals: Adding options for "Every X hours" (e.g., 6h, 12h, 24h) instead of just minutes.

**Comment by diminDDL** [comment]
I guess it's slightly out of scope but it would also be nice if there was also a way for the server to suggest an "automatic" schedule to client apps. This would be useful in cases where the server admin can adjust sync times of clients to be during off peak hours for the server, or when hard drives are spun up for a scrub etc.

**Comment by PANICBUTT0N** [comment]
Seconding this feature request.

**Comment by Brunty** [comment]
I've used PhotoSync (https://www.photosync-app.com) in the past (back when I used to backup photos to Nextcloud) and it allows selection of location, time, only while charging etc for backups, those sorts of options to allow it to not be running backups during the day could be very helpful.

**Comment by crbon** [comment]
iOS conditional background backup would be a great improvement.

Immich already uses `BGAppRefreshTaskRequest` and `BGProcessingTaskRequest`. For processing tasks, iOS supports requirements such as external power and network connectivity, plus `earliestBeginDate`. This is only a lower time limit, so iOS would still decide when to run the task. It does not guarantee an exact time or provide Wi-Fi-only scheduling.

When iOS gives Immich background time, it could check settings such as:

* Backup window, e.g. 22:00–07:00
* Require external power
* Wi-Fi only
* Minimum time since the previous backup

If the conditions are not met, Immich could skip starting, resuming, or queueing backup work and schedule another attempt for the next permitted period.

The same rules may also need to apply to queued background uploads, because they can continue after the background task finishes.

This would not try to make iOS run Immich at 02:00. iOS would decide when background execution is available, while Immich would decide whether that opportunity is suitable for backup.

**Comment by Quentame** [comment]
For iOS : would it be possible to do so via the Shortcut app ?

But Immich have to expose Shortcut commands first.
